test/Microsoft.AspNetCore.Sockets.Tests/TestWebSocketConnectionFeature.cs

```diff
@@ -107,23 +108,45 @@ public override void Dispose()
             {
                 try
                 {
-                    await _input.WaitToReadAsync();
-
-                    if (_input.TryRead(out var message))
+                    if (_internalBuffer.Buffer == null || _internalBuffer.Buffer.Length == 0)
                     {
-                        if (message.MessageType == WebSocketMessageType.Close)
+                        await _input.WaitToReadAsync();
+
+                        WebSocketMessage message;
+                        if (_input.TryRead(out message))
                         {
-                            _state = WebSocketState.CloseReceived;
-                            _closeStatus = message.CloseStatus;
-                            _closeStatusDescription = message.CloseStatusDescription;
-                            return new WebSocketReceiveResult(0, WebSocketMessageType.Close, true, message.CloseStatus, message.CloseStatusDescription);
+                            if (message.MessageType == WebSocketMessageType.Close)
+                            {
+                                _state = WebSocketState.CloseReceived;
+                                _closeStatus = message.CloseStatus;
+                                _closeStatusDescription = message.CloseStatusDescription;
+                                return new WebSocketReceiveResult(0, WebSocketMessageType.Close, true, message.CloseStatus, message.CloseStatusDescription);
+                            }
+
+                            _internalBuffer = message;
                         }
+                    }
 
-                        // REVIEW: This assumes the buffer passed in is > the buffer received
```

Comment:
Just to clarify, it's because of this that we have to update this class right? Because we now pass in a `0`-byte buffer?

Reply:
Yeah :(

Reply:
@BrennanConroy does this code work? If it's tested enough then it's fine.

Reply:
All our non-functional websockets tests use it

Reply:
Does it work? DOES IT WORK? Of course it works.